src/prompts/password.rs

```diff
@@ -322,12 +359,21 @@ impl<'a> PasswordPrompt<'a> {
 
         match self.display_mode {
             PasswordDisplayMode::Hidden => {
+                if self.confirmation_first_answer.is_some() {
+                    backend.render_prompt(self.main_message)?;
+                }
                 backend.render_prompt(prompt)?;
```

Comment:
```suggestion
                backend.render_prompt(&self.message)?;
                match self.confirmation {
                  Some(confirmation) if self.confirmation_stage => backend.render_prompt(confirmation.message),
                  _ => {},
                }
```

doesn´t look too pretty tbh, but iirc `if var && let Some(val) = optional {}` is not stabilized yet.

anyway, just dropping this comment to think how this would like with the confirmation prompt + flag

Reply:
I solved it a bit differently, since we can't have a common prestep for the main prompt (since it would overtake focus otherwise, regardless of stage). See my new code changes here to see if this structure is suitable.